controllers/mdutil/util.go

```diff
@@ -193,7 +193,7 @@ func getIntFromAnnotation(ms *clusterv1.MachineSet, annotationKey string, logger
 	if !ok {
 		return int32(0), false
 	}
-	intValue, err := strconv.Atoi(annotationValue)
+	intValue, err := strconv.ParseInt(annotationValue, 10, 64)
```

Comment:
```suggestion
	desired, err := strconv.ParseInt(desiredString, 10, strconv.IntSize)
```

Though it looks like we are casting below to an int32, maybe it should actually be 32

Reply:
Good catch